framework/math/ParallelVector/ParallelVector.h

```diff
@@ -110,27 +134,16 @@ class ParallelVector
   virtual void operator+=(const ParallelVector& y) = 0;
 
   /**Adds a vector multiplied by scalar a. Optimized for a=1.0 and -1.0*/
-  virtual void PlusAY(double a, const ParallelVector& y) = 0;
+  virtual void PlusAY(const ParallelVector& y, double a) = 0;
 
-  /**Sets the local values of one vector equal to another. The sizes must be
-   * compatible.*/
-  virtual void CopyValues(const ParallelVector& y) = 0;
+  /**Performs x = a*x + y with the current vector being x.*/
+  virtual void AXPlusY(double a, const ParallelVector& y) = 0;
 
-  /**Copies a contiguous block of local data (num_values entries) from the
-  * source vector (starting at y_offset) to the
-   * current vector starting at local_offset. */
-  virtual void BlockCopyLocalValues(const ParallelVector& y,
-                                    int64_t y_offset,
-                                    int64_t local_offset,
-                                    int64_t num_values) = 0;
+  /**Scales a vector by a scalar*/
+  virtual void Scale(double a) = 0;
 
-  /**Copies a contiguous block of data from the source STL vector to the
-   * current vector starting at local_offset. The input STL vector must have
-   * exactly num_values entries.
-   * */
-  virtual void BlockCopyLocalValues(const std::vector<double>& y,
-                                    int64_t local_offset,
-                                    int64_t num_values) = 0;
+  /**Adds a constant scalar value to all the entries of the vector*/
```

Comment:
A little more necessary here than for Scale, but still not really. Consider adding the simple equation to the doc.